job.go

```diff
@@ -108,9 +110,9 @@ func (j *Job) ScheduledAtTime() string {
 
 // Weekday returns which day of the week the Job will run on and
 // will return an error if the Job is not scheduled weekly
-func (j *Job) Weekday() (time.Weekday, error) {
-	if j.unit == weeks {
-		return j.startDay, nil
+func (j *Job) Weekday() (*time.Weekday, error) {
+	if j.scheduledWeekday == nil {
+		return nil, ErrNotScheduledWeekday
 	}
-	return time.Sunday, ErrNotScheduledWeekday
+	return j.scheduledWeekday, nil
```

Comment:
```suggestion
	return *j.scheduledWeekday, nil
```

Reply:
I see. We'll still need the check, though. Users may call Weekday on a non-weekday job and that'd panic. An workaround for this is setting a default weekday that'll be ignored (which is our current solution actually). What do you think?

Reply:
Yes, we will need the check for sure.

Reply:
done